### PR TITLE
Image IR refactor

### PR DESCRIPTION
**Pull request by kvark**
Improves our handling of images in the IR, adds the "imageLoad" expression.

### CODE REVIEW

File: src/front/spv/mod.rs [hunk 1]
```diff
@@ -817,18 +817,16 @@ impl<I: Iterator<Item = u32>> Parser<I> {
                     match type_arena[image_type_handle].inner {
                         //TODO: compare the result type
                         crate::TypeInner::Image {
-                            base: _,
+                            kind: _,
```
Comment (Napokue) [inline src/front/spv/mod.rs:820]: Nit: can't this be changed to using the syntax with dots? e.g. :

```rust
crate::TypeInner::Image {
    dim,
    arrayed,
    class: crate::ImageClass::Sampled,
    ..
} => {
```

File: src/front/spv/mod.rs [hunk 2]
```diff
@@ -877,7 +876,12 @@ impl<I: Iterator<Item = u32>> Parser<I> {
                     };
                     match type_arena[image_type_handle].inner {
                         //TODO: compare the result type
-                        crate::TypeInner::DepthImage { dim, arrayed } => {
+                        crate::TypeInner::Image {
+                            kind: _,
```
Comment (Napokue) [inline src/front/spv/mod.rs:880]: Same here

**Review by Napokue — CHANGES_REQUESTED** [review]
Thank you! Overall looking good, only the front/glsl_new/parser.rs has quite a lot problems with clippy it seems. Got one nitpick comment, rest LGTM.

File: src/front/spv/mod.rs (hunk 1 above)
Comment (kvark) [reply src/front/spv/mod.rs:820]: I believe explicit syntax `kind: _` is superior. If new fields are added to the struct, the author would have to revisit these places and decided if these fields need to be considered.

**Comment by kvark** [comment]
All should be fixed now. Added a clippy setting to allow the explicit pattern.

**Pull request merged**